- name: Cleaning the exercise2
  hosts: localhost
  gather_facts: true

  vars:
    user: "{{ lookup('env', 'USER') }}"

  tasks:
    - block:
        - name: Try to fetch namespaces to check connection
          kubernetes.core.k8s_info:
            api_version: v1
            kind: Namespace
          register: namespaces
          no_log: true
      rescue:
        - name: Display error message
          ansible.builtin.fail:
            msg: "You are not connected into the cluster. Please use the lab instruction to connect on the cluster using user '{{ user }}'"

    - name: Breaking the exercise2
      kubernetes.core.k8s:
        state: absent
        template: "exercise2-fix-vm.yaml.j2"
      register: breakvm
      until: breakvm is succeeded
      retries: 60
      delay: 10
      failed_when: breakvm is failed or breakvm.failed
